test/Directory.Build.props

```diff
@@ -7,6 +7,7 @@
     <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('OSX')) AND '$(TargetArchitecture)' == 'arm64'">net6.0</TargetFrameworks>
     <TargetFrameworks Condition="'$(TestTargetFramework)' != ''">$(TestTargetFramework)</TargetFrameworks>
     <RuntimeIdentifier Condition="'$(TargetFramework)' == 'net461'">win-x64</RuntimeIdentifier>
+    <IsTestProject>true</IsTestProject>
```

Comment:
Previously `xunit` was referenced transitively by `Microsoft.DotNet.XUnitExtensions`.  That's no longer the case, though some XUnit packages are referenced.  In particular `xunit.core`.  

It sets `IsTestProject` to true which will cause https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.Arcade.Sdk/tools/Tests.targets#L83 to import Xunit targets, but only *after* a restore is done.  As a result those targets don't actually reference the xunit package at the right time.  We'll get compile errors due to missing references.  I could add a `PackageReference` here, but then that will trigger a warning from the SDK because it sees the implicit reference from arcade (that was actually never restored).

The fix is to set `IsTestProject` since this is the `Arcade` contract.